src/protocol/uniswap_v2/state.rs

```diff
@@ -93,22 +94,22 @@ impl ProtocolSim for UniswapV2State {
         amount_in: U256,
         token_in: &ERC20Token,
         token_out: &ERC20Token,
-    ) -> Result<GetAmountOutResult, TradeSimulationError> {
+    ) -> Result<GetAmountOutResult, TychoSimulationError> {
         if amount_in == U256::zero() {
-            return Result::Err(TradeSimulationError::new(
+            return Result::Err(TychoSimulationError::from(NativeSimulationError::new(
```

Comment:
oh I would expect that we don't have to explicitly convert this like this 😬 
(I checked locally and I see we actually do buuuuuh)